tests/ngff/test_ngff.py

```diff
@@ -645,6 +646,70 @@ def test_set_scale(ch_shape_dtype):
             assert dataset.zattrs["iohub"]["prior_z_scale"] == 3.0
 
 
+@given(
+    ch_shape_dtype=_channels_and_random_5d_shape_and_dtype(),
+)
+@settings(max_examples=16, deadline=2000)
+def test_set_contrast_limits(ch_shape_dtype):
+    """Test `iohub.ngff.Position.set_contrast_limits()`"""
+    channel_names, shape, dtype = ch_shape_dtype
+    # Skip if no channels or only one channel
+    assume(len(channel_names) > 1)
+
+    with TemporaryDirectory() as temp_dir:
+        store_path = os.path.join(temp_dir, "ome.zarr")
+        dataset = open_ome_zarr(
+            store_path, layout="fov", mode="a", channel_names=channel_names
+        )
+        dataset.create_zeros("data", shape, dtype)
+
+        # Store the initial window settings for all channels
+        initial_windows = {}
+        for ch_name in channel_names:
+            ch_idx = dataset.get_channel_index(ch_name)
+            initial_windows[ch_name] = dataset.metadata.omero.channels[
+                ch_idx
+            ].window
+
+        # Test setting contrast limits for the first channel only
+        target_channel = channel_names[0]
+        window = {"start": 10.0, "end": 100.0, "min": 0.0, "max": 255.0}
+
+        # Set contrast limits
+        dataset.set_contrast_limits(target_channel, window)
+
+        # Check that he contrast tlimits were set
+        # correctly for the target channel
+        channel_index = dataset.get_channel_index(target_channel)
+        assert (
+            dataset.metadata.omero.channels[channel_index].window is not None
```

Comment:
`dataset.metadata.omero.channels[channel_index].window` is repeated 5 times.

Reply:
is this better?